pkg/kube/activity.go

```diff
@@ -161,8 +163,23 @@ func GenerateBuildNumber(activities typev1.PipelineActivityInterface, pipelines
 	return build, answer, nil
 }
 
+func lazyUpdateSourceRepositoryCRD(jxClient versioned.Interface, ns string, activity *v1.PipelineActivity) error {
+	//FIXME: this relies on a bad assumption that the name of the sourcerepositories CRD entry is always equal to the repository name
+	_, err := sourcerepository.NewSourceRepositoryService(jxClient, ns).GetSourceRepository(activity.RepositoryName())
+
+	if err != nil {
+		// sourcerepository for this project does not exist yet, so create one
+		err := sourcerepository.NewSourceRepositoryService(jxClient, ns).CreateSourceRepository(
+			activity.RepositoryName(), activity.Spec.GitOwner,activity.Spec.GitURL)
```

Comment:
Nit: Space after the comment in parameter lists
```suggestion
			activity.RepositoryName(), activity.Spec.GitOwner, activity.Spec.GitURL)
```